ultra/ultra/train.py

```diff
@@ -45,24 +45,77 @@
 num_gpus = 1 if torch.cuda.is_available() else 0
 
 
-def train(
-    scenario_info,
-    num_episodes,
-    policy_classes,
-    max_episode_steps,
-    max_steps,
-    eval_info,
-    timestep_sec,
-    headless,
-    seed,
-    log_dir,
-    policy_ids=None,
-):
-    torch.set_num_threads(1)
-    total_step = 0
-    finished = False
-    evaluation_task_ids = dict()
+def create_parser():
+    parser = argparse.ArgumentParser("intersection-training")
+    parser.add_argument(
+        "--task", help="Tasks available : [0, 1, 2]", type=str, default="1"
+    )
+    parser.add_argument(
+        "--level",
+        help="Levels available : [easy, medium, hard, no-traffic]",
+        type=str,
+        default="easy",
+    )
+    parser.add_argument(
+        "--policy",
+        help="Policies available : [ppo, sac, td3, dqn, bdqn]",
+        type=str,
+        default="sac",
+    )
+    parser.add_argument(
+        "--episodes", help="Number of training episodes", type=int, default=1000000
+    )
+    parser.add_argument(
+        "--max-episode-steps",
+        help="Maximum number of steps per episode",
+        type=int,
+        default=200,
+    )
+    parser.add_argument(
+        "--max-steps",
+        help="Maximum total number of training steps",
+        type=int,
+        default=1000000,
+    )
+    parser.add_argument(
+        "--timestep", help="Environment timestep (sec)", type=float, default=0.1
+    )
+    parser.add_argument(
+        "--headless",
+        help="Run without envision",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--eval-episodes", help="Number of evaluation episodes", type=int, default=200
+    )
+    parser.add_argument(
+        "--eval-rate",
+        help="The number of training episodes to wait before running the evaluation",
+        type=int,
+        default=200,
+    )
+    parser.add_argument(
+        "--seed",
+        help="Environment seed",
+        default=2,
+        type=int,
+    )
+    parser.add_argument(
+        "--log-dir",
+        help="Log directory location",
+        default="logs",
+        type=str,
+    )
+    parser.add_argument(
+        "--policy-ids",
+        help="Name of each specified policy",
+        default=None,
+        type=str,
+    )
+    return parser
+
 
+def build_agent(policy_classes, policy_ids, max_episode_steps):
```

Comment:
```suggestion
def build_agents(policy_classes, policy_ids, max_episode_steps):
```
(And then change calls to this function if we go with this change.)